clang/include/clang/Basic/DiagnosticParseKinds.td

```diff
@@ -1517,4 +1517,7 @@ def err_expected_expr_in_where_clause : Error<
 def err_invalid_decl_where_clause : Error<
   "where clause on invalid decl">;
 
+def err_invalid_stmt_where_clause : Error<
+  "where clause attached to invalid statement">;
```

Comment:
This error message is emitted on line 518 of ParseStmt.cpp if the `StmtRes` is not a `ValueStmt`. In practice, I don't see a scenario where  an `ExprStmt` is not a `ValueStmt`. So this error message may never be actually emitted.
An alternative is to assert/llvm_unreachable instead of emitting this error.